core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -981,67 +981,62 @@ class WorkspaceService(protected val ctx: RawlsRequestContext,
-                        workspaceSamResource.authDomainGroups.map(groupName =>
-                          ManagedGroupRef(RawlsGroupName(groupName.value))
-                        )
-                      ),
-                      attributesEnabled,
-                      wsmContext.getCloudPlatform
-                    )
-                  // remove submission stats if they were not requested
-                  val submissionStats: Option[WorkspaceSubmissionStats] = if (submissionStatsEnabled) {
-                    Option(submissionSummaryStats(wsId))
+              aggregatedWorkspaces.mapFilter { wsmContext =>
+                val workspace = wsmContext.baseWorkspace
+                val wsId = UUID.fromString(workspace.workspaceId)
+                val workspaceSamResource = workspaceSamResourceByWorkspaceId(workspace.workspaceId)
+                val accessLevel =
+                  if (workspaceSamResource.missingAuthDomainGroups.nonEmpty) {
+                    WorkspaceAccessLevels.NoAccess
                   } else {
-                    None
+                    highestAccessLevelByWorkspaceId.getOrElse(workspace.workspaceId, WorkspaceAccessLevels.NoAccess)
                   }
-                  // Remove workspaces that are non-ready with no cloud context (Ready workspaces with no
-                  // cloud context will throw a WorkspaceAggregationException, which is handled below)
-                  wsmContext.getCloudPlatform match {
-                    case None => None
-                    case _ =>
-                      Option(
-                        WorkspaceListResponse(
-                          accessLevel,
-                          workspaceDetails,
-                          submissionStats,
-                          workspaceSamResource.public.roles.nonEmpty || workspaceSamResource.public.actions.nonEmpty,
-                          Some(wsmContext.policies)
-                        )
+
+                // remove attributes if they were not requested
+                val workspaceDetails =
+                  WorkspaceDetails.fromWorkspaceAndOptions(
+                    workspace,
+                    Option(
+                      workspaceSamResource.authDomainGroups.map(groupName =>
+                        ManagedGroupRef(RawlsGroupName(groupName.value))
+                      )
+                    ),
+                    attributesEnabled,
+                    wsmContext.getCloudPlatform
+                  )
+                // remove submission stats if they were not requested
+                val submissionStats: Option[WorkspaceSubmissionStats] = if (submissionStatsEnabled) {
+                  Option(submissionSummaryStats(wsId))
+                } else {
+                  None
+                }
+                // Remove workspaces that are non-ready with no cloud context (Ready workspaces with no
+                // cloud context will throw a WorkspaceAggregationException, which is handled below)
```

Comment:
Where is the WorkspaceAggregationException handled?

Reply:
It's logged and filtered (same as existing behavior), in the `recover` clause.